tests/test_clients_OpenstackClient.py

```diff
@@ -0,0 +1,186 @@
+        self.patchers.append(create_start_patcher(
+            patch_function='keystoneauth1.identity.v3.Password'))
+        self.patchers.append(create_start_patcher(
+            patch_function='keystoneauth1.session.Session'))
+        self.patchers.append(create_start_patcher(
+            patch_function='get_project_id', patch_object=KeystoneSession))
+        self.patchers.append(create_start_patcher(
+            patch_function='create_cinder_client', patch_object=OpenstackClient, return_value=CinderClient()))
+        self.patchers.append(create_start_patcher(
+            patch_function='create_nova_client', patch_object=OpenstackClient, return_value=NovaClient()))
+        self.patchers.append(create_start_patcher(
+            patch_function='create_swift_client', patch_object=OpenstackClient, return_value=SwiftClient()))
+        self.patchers.append(create_start_patcher(
+            patch_function='create_swift_service', patch_object=OpenstackClient, return_value=SwiftService()))
+
+        os.environ['SF_BACKUP_RESTORE_LOG_DIRECTORY'] = log_dir
+        os.environ['SF_BACKUP_RESTORE_LAST_OPERATION_DIRECTORY'] = log_dir
+
+        self.osClient = OpenstackClient(operation_name, configuration, directory_persistent, directory_work_list,
+                                        poll_delay_time, poll_maximum_time)
+
+    @classmethod
+    def teardown_class(self):
+        stop_all_patchers(self.patchers)
+
+    def test_create_os_client(self):
+        assert isinstance(self.osClient.nova, NovaClient)
+        assert isinstance(self.osClient.cinder, CinderClient)
+        assert isinstance(self.osClient.swift, SwiftClient)
+        assert isinstance(self.osClient.swift.service, SwiftService)
+        assert self.osClient.availability_zone == availability_zone
+        assert self.osClient.container == file_to_dict(
+            'tests/data/openstack/swift.head_container.txt')
+
+    def test_get_container_exception(self):
+        self.osClient.CONTAINER = invalid_container
+        assert self.osClient.get_container() is None
+        self.osClient.CONTAINER = valid_container
+
+    def test_get_snapshot(self):
+        expected_snapshot = Snapshot(valid_snapshot_name, 40, 'available')
+        snapshot = self.osClient.get_snapshot(valid_snapshot_name)
+        assert expected_snapshot.id == snapshot.id
+        assert expected_snapshot.size == snapshot.size
+        assert expected_snapshot.status == snapshot.status
+
+    def test_get_snapshot_exception(self):
+        assert self.osClient.get_snapshot(not_found_snapshot_name) is None
+
+    def test_get_volume(self):
+        expected_volume = Volume(valid_disk_name, 'in-use', 40)
+        volume = self.osClient.get_volume(valid_disk_name)
+        assert expected_volume.id == volume.id
+        assert expected_volume.size == volume.size
+        assert expected_volume.status == volume.status
+
+    def test_get_volume_exception(self):
```

Comment:
Should we also add test cases for delete_volume or delete_snapshot?

Reply:
Yes @ketakig12 .. I had requested mainly for delete functions since those were the changes made in this PR :)